snap/local/start-exporter.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+set -eo pipefail # Exit on error
+
+EXPORTER_OPTS=""
+EXPORTER_PATH="/usr/bin/prometheus-postgres-exporter"
+SOCKET_PATH="/tmp"
+
+if [ -z "$SNAP" ]; then
+    # When not running as a snap, expect `DATA_SOURCE_NAME` to be set.
+    if [ -z "$DATA_SOURCE_NAME" ]; then
+        echo "DATA_SOURCE_NAME must be set"
+        exit 1
+    fi
+    exec "$EXPORTER_PATH" $(echo "$EXPORTER_OPTS")
+else
+    # When running as a snap, expect `exporter.user` and `exporter.password`
+    EXPORTER_USER="$(snapctl get exporter.user)"
+    EXPORTER_PASS="$(snapctl get exporter.password)"
+    EXPORTER_PATH="${SNAP}${EXPORTER_PATH}"
+
+    if [[ -z "$EXPORTER_USER" || -z "$EXPORTER_PASS" ]]; then
+        echo "exporter.user and exporter.password must be set"
```

Comment:
Apologies for the old-schooling here. Errors must be printed to STDERR (good to indicate it is an 'error' too):
```suggestion
        echo "Error: exporter.user and exporter.password must be set" 2>&1
```

Reply:
2b613a4